tracer/test/Datadog.Trace.TestHelpers.AutoInstrumentation/TestHelper.cs

```diff
@@ -644,6 +646,62 @@ protected void EnableDirectLogSubmission(int intakePort, string integrationName,
             Assert.Equal(expectedServiceVersion, span.Tags.GetValueOrDefault(Tags.Version));
         }
 
+        protected async Task ReportRetry(ITestOutputHelper outputHelper, int attemptsRemaining, Type testType, Exception ex = null)
+        {
+            var type = outputHelper.GetType();
+            var testMember = type.GetField("test", BindingFlags.Instance | BindingFlags.NonPublic);
+            var test = (ITest)testMember?.GetValue(outputHelper);
+            var testFullName = type.FullName + test?.TestCase.DisplayName;
+
+            outputHelper.WriteLine($"Error executing test: {testFullName}. {attemptsRemaining} attempts remaining. {ex}");
+
+            // In addition to logging, send a metric that will help us get more information through tags
+            var srcBranch = Environment.GetEnvironmentVariable("DD_LOGGER_BUILD_SOURCEBRANCH");
+
+            var tags = $$"""
+                "os.platform:{{SanitizeTagValue(FrameworkDescription.Instance.OSPlatform)}}",
+                "os.architecture:{{SanitizeTagValue(EnvironmentTools.GetPlatform())}}",
+                "target.framework:{{SanitizeTagValue(EnvironmentHelper.GetTargetFramework())}}",
+                "test.name:{{SanitizeTagValue(testFullName)}}",
+                "git.branch:{{SanitizeTagValue(srcBranch)}}"
+            """;
+
+            var payload = $$"""
+                {
+                "series": [
+                {
+                    "metric": "dd_trace_dotnet.ci.tests.retries",
+                    "type": 1,
+                    "points": [
+                    {
+                        "timestamp": {{((DateTimeOffset)DateTime.UtcNow).ToUnixTimeSeconds()}},
+                        "value": 1
+                    }
+                    ],
+                    "tags": [
+                        {{tags}}
+                    ]
+                }
+                ]
+            });
```

Comment:
This isn't valid json 🤔 I assume it's meant to be this, but personally I would inline the `tags` variable too
```suggestion
                {
                    "series": [{
                        "metric": "dd_trace_dotnet.ci.tests.retries",
                        "type": 1,
                        "points": [{
                            "timestamp": {{((DateTimeOffset)DateTime.UtcNow).ToUnixTimeSeconds()}},
                            "value": 1
                        }],
                        "tags": {
                            {{tags}}
                        }
                    }]
                }
```

Reply:
The tags are supposed to be an array of string: https://docs.datadoghq.com/api/latest/metrics/#submit-metrics

Reply:
Thanks, the funny thing was that I had tested and it worked, the backend accepts the malformed JSON. 
So at first I was puzzled by your comment even though it's clearly not a valid json :) Fixed in df6e341ff1bcb719b638a0990ebd19810011ba54